addon/bootstrap.js

```diff
@@ -40,133 +35,117 @@ const REASONS = {
-  Jsm.unload([CONFIGPATH, STUDYUTILSPATH, RAPPORPATH]);
+  Jsm.unload([CONFIGPATH, STUDY_UTILS_PATH, HOMEPAGE_STUDY_PATH]);
 }
 
 function shutdown(addonData, reason) {
-  console.log("shutdown", REASONS[reason] || reason);
-  // are we uninstalling?
-  // if so, user or automatic?
+  log.debug("shutdown", REASONS[reason] || reason);
+  // Are we uninstalling? if so, user or automatic?
   if (reason === REASONS.ADDON_UNINSTALL || reason === REASONS.ADDON_DISABLE) {
-    console.log("uninstall or disable");
+    log.debug("uninstall or disable");
     if (!studyUtils._isEnding) {
-      // we are the first requestors, must be user action.
-      console.log("user requested shutdown");
+      // We are the first requestors, must be user action.
+      log.debug("user requested shutdown");
       studyUtils.endStudy({reason: "user-disable"});
     }
   }
   unload();
 }
 
 function uninstall(addonData, reason) {
-  console.log("uninstall", REASONS[reason] || reason);
+  log.debug("uninstall", REASONS[reason] || reason);
 }
 
 function install(addonData, reason) {
-  console.log("install", REASONS[reason] || reason);
-  // handle ADDON_UPGRADE (if needful) here
-}
-
-// logging
-function createLog(name, levelWord) {
-  Cu.import("resource://gre/modules/Log.jsm");
-  var L = Log.repository.getLogger(name);
-  L.addAppender(new Log.ConsoleAppender(new Log.BasicFormatter()));
-  L.level = Log.Level[levelWord] || Log.Level.Debug; // should be a config / pref
-  return L;
-}
-
-async function chooseVariation() {
-  let toSet, source;
-  const sample = studyUtils.sample;
-
-  if (studyConfig.variation) {
-    source = "startup-config";
-    toSet = studyConfig.variation;
-  } else {
-    source = "weightedVariation";
-    // this is the standard arm choosing method
-    const clientId = await studyUtils.getTelemetryId();
-    const hashFraction = await sample.hashFraction(studyConfig.studyName + clientId, 12);
-    toSet = sample.chooseWeighted(studyConfig.weightedVariations, hashFraction);
-  }
-  log.debug(`variation: ${toSet} source:${source}`);
-  return toSet;
-}
-
-function getHomepage(){
-  // get the homepage of the user
-  var homepage = Services.prefs.getComplexValue(PREF_HOMEPAGE, Ci.nsIPrefLocalizedString).data;
-  // transform the homepage into a nsIURI. Neccesary to get the base domain
-  var homepageURI = Services.netUtils.newURI(homepage);
-
-  var eTLD;
-  if (homepage.startsWith("about:")) {
-    // If the homepage starts with 'about:' (see about:about)
-    eTLD = "about:pages";
-  } else {
-    try {
-      eTLD = Services.eTLD.getBaseDomain(homepageURI);
-    } catch (e) {
-      // getBaseDomain will fail if the host is an IP address or is empty
-      eTLD = homepage;
-    }
-  }
-  return eTLD;
-}
\ No newline at end of file
+  // NOTE: the registered chrome url is not available in the install phase,
+  // it is only available once the addon has been started.
+  //log.debug("install", REASONS[reason] || reason);
+ }
```

Comment:
Please add a newline at the end of this file.

Reply:
Done